lib/index.ts

```diff
@@ -571,4 +574,40 @@ export default class WalletClient {
       this.eventListeners.get(event)!.forEach((listener) => listener(data));
     }
   }
+
+  private populateWindowObject() {
+    if (typeof window !== 'undefined') {
+      const createMethodWrapper = (method: Function) => {
+        return async (...args: any[]) => {
+          if (!this.isConnected) {
+            throw new Error("Wallet is not connected. Please call connect() first.");
+          }
+          return method.apply(this, args);
+        };
+      };
+
+      const walletApi: any = {
```

Comment:
Again avoid having any. After creation of the type also export that for libs user, so he can make something like

```ts
window as TypeCreatedForAPI
```

Reply:
not sure if works for window global object, but may be TS namespace can be created and window described there, so using window TS will automatically understand the type of patched window